param/parameterized.py

```diff
@@ -2504,9 +2504,33 @@ def type_script_repr(type_,imports,prefix,settings):
 dbprint_prefix=None
 
 
+# Copy of Python 3.2 reprlib's recursive_repr but allowing extra arguments
+if sys.version_info.major >= 3:
+    from threading import get_ident
+    def recursive_repr(fillvalue='...'):
+        'Decorator to make a repr function return fillvalue for a recursive call'
+
+        def decorating_function(user_function):
+            repr_running = set()
+
+            def wrapper(self, *args, **kwargs):
+                key = id(self), get_ident()
+                if key in repr_running:
+                    return fillvalue
+                repr_running.add(key)
+                try:
+                    result = user_function(self, *args, **kwargs)
+                finally:
+                    repr_running.discard(key)
+                return result
+            return wrapper
 
-
-
+        return decorating_function
+else:
+    def recursive_repr(fillvalue='...'):
+        def decorating_function(user_function):
+            return user_function
+        return decorating_function
```

Comment:
Couldn't be bothered? 😆

Reply:
To fix this issue on py2? Definitely not!